doc/examples/Webhook.rst

```diff
@@ -0,0 +1,112 @@
+'pull_request' and 'push' for the EVENT attributes, any time a PR is opened, closed, merged, or synced, or a commit is pushed,
+Github sends a POST containing a payload with information about the PR/push and its state.
+
+The below example was drawn largely from `Github's Examples <https://github.com/github/platform-samples/blob/master/api/python/building-a-ci-server/server.py>`__
+on working with Webhooks. A list of all applicable event types for Webhooks can be found in `Github's documentation <https://developer.github.com/v3/issues/events/>`__
+
+.. code-block:: python
+
+    from __future__ import print_function
+
+    from wsgiref.simple_server import make_server
+    from pyramid.config import Configurator
+    from pyramid.view import view_config, view_defaults
+    from pyramid.response import Response
+    from github import Github
+
+    ENDPOINT = "webhook"
+
+    @view_defaults(
+        route_name=ENDPOINT, renderer="json", request_method="POST"
+    )
+    class PayloadView(object):
+        """
+        View receiving of Github payload. By default, this view it's fired only if
+        the request is json and method POST.
+        """
+
+        def __init__(self, request):
+            self.request = request
+            # Payload from Github, it's a dict
+            self.payload = self.request.json
+
+        @view_config(header="X-Github-Event:push")
+        def payload_push(self):
+            """This method is a continuation of PayloadView process, triggered if
+            header HTTP-X-Github-Event type is Push"""
+            print("No. commits in push:", len(self.payload['commits']))
+            return Response("success")
+
+        @view_config(header="X-Github-Event:pull_request")
+        def payload_pull_request(self):
+            """This method is a continuation of PayloadView process, triggered if
+            header HTTP-X-Github-Event type is Pull Request"""
+            print("PR", self.payload['action'])
+            print("No. Commits in PR:", self.payload['pull_request']['commits'])
+
+            return Response("success")
+
+        @view_config(header="X-Github-Event:ping")
+        def payload_else(self):
+            print("Pinged! Webhook created with id {}!".format(self.payload["hook"]["id"]))
+            return {"status": 200}
+
+
+    def create_webhook():
+        USERNAME = ""
+        PASSWORD = ""
+        OWNER = ""
+        REPO_NAME = ""
+        EVENTS = ["push", "pull_request"]
+        HOST = ""
+
+        config = {
+            "url": "http://{host}/{endpoint}".format(host=HOST, endpoint=ENDPOINT),
+            "content_type": "json"
+        }
+
+        g = Github(USERNAME, PASSWORD)
+        repo = g.get_repo("{owner}/{repo_name}".format(owner=OWNER, repo_name=REPO_NAME))
+        repo.create_hook("web", config, EVENTS, active=True)
+
+
+    if __name__ == "__main__":
+        config = Configurator()
+
+        create_webhook()
```

Comment:
Maybe you can add a mention this step can also be done from Github itself. (under repo settings)